cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,1397 @@
+  // at least 16
+  ASSERT_THROW(ContentDefinedChunker(li, 0, -1), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 1024, 512), ParquetException);
+
+  ASSERT_THROW(ContentDefinedChunker(li, -1, 0), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 0, 0), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, 16));
+  ASSERT_THROW(ContentDefinedChunker(li, -16, -16), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 16, 0), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 32, 32), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 32, 48));
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 1024 * 1024, 2 * 1024 * 1024));
+  ASSERT_NO_THROW(
+      ContentDefinedChunker(li, 1024 * 1024 * 1024L, 2LL * 1024 * 1024 * 1024L));
+
+  // with norm_factor=1 the difference between min and max chunk size must be
+  // at least 64
+  ASSERT_THROW(ContentDefinedChunker(li, 1, -1, 1), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, -1, 1, 1), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 1, 1, 1), ParquetException);
+  ASSERT_THROW(ContentDefinedChunker(li, 1, 32, 1), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 1, 33, 1));
+
+  // with norm_factor=2 the difference between min and max chunk size must be
+  // at least 128
+  ASSERT_THROW(ContentDefinedChunker(li, 0, 63, 2), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, 64, 2));
+
+  // with norm_factor=-1 the difference between min and max chunk size must be
+  // at least 8
+  ASSERT_THROW(ContentDefinedChunker(li, 0, 7, -1), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, 8, -1));
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, 16, -1));
+
+  // test the norm_factor extremes
+  ASSERT_THROW(ContentDefinedChunker(li, 0, 0, -68), ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, 1, -67));
+  ASSERT_THROW(ContentDefinedChunker(li, 0, std::numeric_limits<int64_t>::max(), 59),
+               ParquetException);
+  ASSERT_NO_THROW(ContentDefinedChunker(li, 0, std::numeric_limits<int64_t>::max(), 58));
+}
+
+class TestCDCSingleRowGroup : public ::testing::TestWithParam<CaseConfig> {
+ protected:
+  // Column random table parts for testing
+  std::shared_ptr<Field> field_;
+  std::shared_ptr<Table> part1_, part2_, part3_, part4_, part5_, part6_, part7_;
+
+  void SetUp() override {
+    const auto& param = GetParam();
+    auto field_ = ::arrow::field("f0", param.dtype, param.is_nullable);
+    auto schema = ::arrow::schema({field_});
+
+    auto part_length = kPartSize / param.bytes_per_record;
+    auto edit_length = kEditSize / param.bytes_per_record;
+    ASSERT_OK_AND_ASSIGN(part1_, GenerateTable(schema, part_length, 0));
+    ASSERT_OK_AND_ASSIGN(part2_, GenerateTable(schema, edit_length, 1));
+    ASSERT_OK_AND_ASSIGN(part3_, GenerateTable(schema, part_length, part_length));
```

Comment:
What is the last parameter? Please make it explicit.
```suggestion
    ASSERT_OK_AND_ASSIGN(part3_, GenerateTable(schema, part_length, /*seed=*/ part_length));
```

Reply:
Added wherever GenerateTable() is being used.